design/one-pager-consuming-k8s-clusters.md

```diff
@@ -0,0 +1,317 @@
+`KubernetesCluster` claims in their namespace. With the introduction of the
+`KubernetesTarget`, `KubernetesApplication` resources can still be scheduled via
+label selectors, but will now be scheduled to a `KubernetesTarget` in their
+namespace, then use its `connectionSecretRef` to get the `Secret` that was
+propagated to the namespace from the Kubernetes cluster managed resource or
+already existed due to manual creation. Importantly, for a
+`KubernetesApplication` to be able to be scheduled to a Kubernetes cluster, a
+`KubernetesTarget` that points to that cluster managed resource or local
+`Secret` *must* be present in the namespace of the `KubernetesApplication`.
+
+This change isolates the actions of *provisioning* (`KubernetesCluster` or
+cloud-specific managed resource), *publishing* (`KubernetesTarget`), and
+*consuming* (`KubernetesApplication`) Kubernetes clusters, actions which may be
+executed by [almost disjoint] sets of users. Having these actions represented by
+separate custom resources allows for more granular [RBAC] permissions around
+Kubernetes cluster usage in Crossplane.
+
+### Automatic KubernetesTarget Creation Controller
+
+Because the ability to *publish* a Kubernetes cluster provides immense power in
+a Crossplane environment, it is desirable for Kubernetes cluster managed
+resources to be able to be consumed in the namespace that they are claimed
+without a user requiring the ability to create a `KubernetesTarget`. For
+instance, if an infrastructure owner wants to allow for an application team to
+provision some clusters dynamically by creating a `KubernetesCluster` claim in
+their namespace that references a `GKEClusterClass`, that application team
+should be able to then schedule `KubernetesApplication` resources to that
+cluster without requiring the permissions to create a `KubernetesTarget` (which
+would allow them to enable to consumption of any Kubernetes cluster managed
+resource) or asking the infrastructure owner to cerate a `KubernetesTarget` for
+them (which would break the self-service model).
+
+To enable this functionality, a single controller should be implemented in core
+Crossplane that watches for the creation of `KubernetesCluster` claim resources
+and automatically creates a `KubernetesTarget` with the `connectionSecretRef`
+set to the connection `Secret` that was propagated to the namespace.
+Importantly, this controller should also clean up the `KubernetesTarget` when
+the `KubernetesCluster` (and corresponding) `Secret` are deleted. This can be
+accomplished by setting an `ownerReference` to the `KubernetesCluster` claim on
+the `KubernetesTarget`. In addition, because this `KubernetesTarget` resource
+may be scheduled to by a `KubernetesApplication` using labels, any labels that
+are present on the `KubernetesCluster` claim should be propagated to the
+`KubernetesTarget` resource and the name of the resource should match that of
+the claim.
```

Comment:
How do we feel about this label propagation and naming strategy for automatically created `KubernetesTarget` resources?

Reply:
I was thinking the same thing - this seems reasonable to me.